File: PlusOne.py
# https://leetcode.com/problems/plus-one/description/
def plusOne(digits: list) -> list:
        nums = ''
        result = []
        for i in digits:
            nums += str(i)
        nums = int(nums) + 1
        return [int(num) for num in str(nums)]

print(plusOne([1,2,3])) # [1,2,4]
print(plusOne([4,3,2,1])) # [4,3,2,2]
print(plusOne([0])) # [1]

File: Arrays&Hashing/PlusOne.java
public class PlusOne {
    public int[] plusOne(int[] digits) {
        int n = digits.length;
        for (int i = n - 1; i >= 0; i--) {
            if (digits[i] < 9) {
                digits[i]++;
                return digits;
            }
            digits[i] = 0;
        }
        int[] answer = new int[n + 1];
        answer[0] = 1;
        return answer;
    }

    public static void main(String[] args) {
        PlusOne solution = new PlusOne();
        System.out.println(solution.plusOne(new int[] {1,2,3}));
    }
}